src/AppInstallerRepositoryCore/SourceList.cpp

```diff
@@ -196,9 +210,9 @@ namespace AppInstaller::Repository
                 Utility::CaseInsensitiveEquals(left.Type, right.Type);
         }
 
-        bool ShouldBeHidden(const SourceDetailsInternal& details)
+        bool ShouldBeHidden(const SourceDetailsInternal& details, bool allowExplicitSources)

```

Comment:
My instinct is that `Explicit` only needs to be handled when constructing the composite source from the source list.  Passing a bool to all of these functions seems very fragile.